src/app.ts

```diff
@@ -71,6 +72,10 @@ async function app() {
       false
     )
     .option('-s, --snapshot', 'generate snapshot in Structure Definition output', false)
+    .option(
+      '-c, --config <config...>',
+      "override elements in sushi-config.yaml (supported: 'version', 'status', 'releaselabel') (eg: --config 'status:draft')"
+    )
```

Comment:
I thought that variadic arguments worked via repeated uses of the option, but it turns out I was wrong. It just expects any number of values after the option. So when I tested this with `build --config status:draft /path/to/my/project` it mistakenly parsed `/path/to/my/project` as one of the config option values instead of as the final path argument.

So in the end, you were right. You need a custom option parser. That said, it's not that difficult. I tried this and it worked well, returning an array of values:

```ts
    .option(
      '-c, --config <config>',
      "override elements in sushi-config.yaml (supported: 'version', 'status', 'releaselabel') (eg: --config 'status:draft')",
      (value: string, previous: string[] = []) => {
        return previous.concat(value);
      }
    )
```

Now that you have a custom option parser already, you could go the next step and put the string parsing in there too and return an object instead:
```ts
    .option(
      '-c, --config <config>',
      "override elements in sushi-config.yaml (supported: 'version', 'status', 'releaselabel') (eg: --config 'status:draft')",
      (value: string, previous = {}) => {
        const [k, ...v] = value.split(':');
        return Object.assign(previous, { [k]: v.join(':') });
      }
    )
```

If you do that then don't forget to change the lines that log options:
```ts
  if (program.config) {
    Object.entries(program.config).forEach(([k, v]) => logger.info(`  --config ${k}:${v}`));
  }
```

And of course that means you can remove the `parseConfigOverrides` function in `Processing.ts` as well (since it would already be parsed into an object). But now you'll have to fix the tests too.

Anyway, it's up to you where you prefer to parse those strings. These are just a couple of options.

Reply:
Yep, you're right. Variadic kind of forces you to use the clumsy `--` syntax.

Also: `const [k, ...v] = value.split(':');` *chef kiss*